arm/proofs/instruction.ml

```diff
@@ -1071,6 +1072,25 @@ let arm_SBCS = define
          CF := (&(val m) - (&(val n) + &c):int = &(val d)) ,,
          VF := ~(ival m - (ival n + &c) = ival d)) s`;;
 
+(* SLI (vector), Q = 1 case only (datasize = 128) *)
+let arm_SLI_VEC = define
+ `arm_SLI_VEC Rd Rn shiftamnt esize =
+    \s. let n:(128)word = read Rn s
+        and d:(128)word = read Rd s in
+        let mask = (2 EXP shiftamnt) - 1 in
+        let d:(128)word = if esize = 64 then
+            simd2 (\ni di. word_or
+              (word_shl ni shiftamnt) (word_and di (word mask))) n d
+          else if esize = 32 then
+            simd4 (\ni di. word_or
+              (word_shl ni shiftamnt) (word_and di (word mask))) n d
+          else if esize = 16 then
+            simd8 (\ni di. word_or
+              (word_shl ni shiftamnt) (word_and di (word mask))) n d
+          else simd16 (\ni di. word_or
+              (word_shl ni shiftamnt) (word_and di (word mask))) n d in
```

Comment:
At some point this big if-then-else pattern should be refactored as well, I believe...